benchmark/buffers/buffer-base64-decode.js

```diff
@@ -0,0 +1,14 @@
+var assert = require('assert');
+var common = require('../common.js');
+
+var bench = common.createBenchmark(main, {});
+
+function main(conf) {
+  for (var s = 'abcd'; s.length < 32 << 20; s += s);
+  assert.equal(s.length % 4, 0);
+  var b = Buffer(s.length / 4 * 3);
+  b.write(s, 0, s.length, 'base64');
+  bench.start();
+  for (var i = 0; i < 32; i += 1) b.write(s, 0, s.length, 'base64');
```

Comment:
I'd suggest using `b.hexWrite(s, 0, s.length)`. It'll be a more direct way to test the C++ changes you've made, and while I'm sure the JIT optimizes it out, the arguments parsing in `.write()` is larger than the actual logic block.